### PR TITLE
[release-1.21] Implement CombinedAnalyzer interface

### PR DESCRIPTION
**Pull request by jjamroga**
**Please provide a description of this PR:**
No new features added, only tweaking interface for extension

Backport, manual cherry pick of https://github.com/istio/istio/pull/50187
Merged 1.20 backport https://github.com/istio/istio/pull/50229

### CODE REVIEW

File: pkg/config/analysis/analyzers/all.go [hunk 1]
```diff
@@ -79,6 +79,14 @@ func All() []analysis.Analyzer {
...
 	return analysis.Combine("all", All()...)
 }
+<<<<<<< HEAD
+=======
+
+// AllMultiClusterCombined returns all multi-cluster analyzers combined as one
+func AllMultiClusterCombined() analysis.CombinedAnalyzer {
+	return analysis.Combine("all-multi-cluster", AllMultiCluster()...)
+}
+>>>>>>> 234f5beb8b (Implement a CombinedAnalyzer interface (#50187))
```
Comment (dhawton) [inline pkg/config/analysis/analyzers/all.go:92]: Bad merge here

Comment (jjamroga) [reply pkg/config/analysis/analyzers/all.go]: Thank you!

**Comment by howardjohn** [comment]
@aryan16 this was merged in 1.20 and master so would be good to get in 1.21 for continuity as well

**Review by zirain — APPROVED** [review]

**Pull request merged**